packages/overlay/src/integrations/sentry/components/traces/spans/SpanTree.tsx

```diff
@@ -21,10 +23,29 @@ export default function SpanTree({
   spanNodeWidth: number;
   setSpanNodeWidth?: (val: number) => void;
 }) {
+  const { query, matchesQuery, showOnlyMatched } = useSearch();
+
+  const filteredTree = useMemo(() => {
+    if (!query) return tree;
+    if (showOnlyMatched) {
+      const spanMemo = new Map<string, boolean>();
+      const hasMatchingDescendant = (span: Span): boolean => {
+        if (spanMemo.has(span.span_id)) return spanMemo.get(span.span_id)!;
+        const result = matchesQuery(span) || (span.children?.some(child => hasMatchingDescendant(child)) ?? false);
+        spanMemo.set(span.span_id, result);
+        return result;
+      };
+
+      return tree.filter(span => hasMatchingDescendant(span));
+    }
+    return tree;
+  }, [query, tree, showOnlyMatched, matchesQuery]);
+
   if (!tree || !tree.length) return null;
```

Comment:
Shouldn't we replace the `tree` here with `filteredTree`?
```suggestion
  if (!filteredTree|| !tree.filteredTree) return null;
```